File: solutions/_349_intersection_of_two_arrays/solution.cpp
#include <set>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> intersection(vector<int> &nums1, vector<int> &nums2)
    {
        set<int> s1(nums1.begin(), nums1.end());
        set<int> s2(nums2.begin(), nums2.end());
        vector<int> result;
        for (auto it = s1.begin(); it != s1.end(); ++it)
        {
            if (s2.find(*it) != s2.end())
            {
                result.push_back(*it);
            }
        }
        return result;
    }
};

/*
https://leetcode.com/problems/intersection-of-two-arrays/
*/

File: solutions/_349_intersection_of_two_arrays/solution.java

package _349_intersection_of_two_arrays;

import java.util.*;

class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<>();
        for (int num : nums1) {
            set1.add(num);
        }

        Set<Integer> set2 = new HashSet<>();
        for (int num : nums2) {
            set2.add(num);
        }

        ArrayList<Integer> result = new ArrayList<>();
        for (int num : set1) {
            if (set2.contains(num)) {
                result.add(num);
            }
        }
        return result.stream().mapToInt(i -> i).toArray();
    }
}

/**
 * https://leetcode.com/problems/intersection-of-two-arrays/
 */